VxLAN Tunnel Counters and Rates implementation

dgsudharsan opened the pull request:
Flexcounter implementation for vxlan tunnel counters and rates implementation

VenkatCisco committed:
Remove redudnant mention of platform cisco-8000 (#856)

Signed-off-by: Venkat Garigipati <venkatg@cisco.com>

dgsudharsan committed:
Changes for tunnel counters

SuvarnaMeenakshi committed:
Revert "[vs]: Start syncd by passing context configuration file and global context index. (#832)" (#859)

This reverts commit 0e2105a8e5c6e14c3551ec2bf65b1ad41bb5060b.
multi-asic VS with multiple swss and syncd dockers, comes up without context_config.json or hwinfo.
Just like a single asic vs, multi-asic vs will spawn multiple instances of swss and syncd with default hwinfo.
Reverting this PR as it is not required currently for multi-asic VS.

kcudnik committed:
[syncd] Fix fdb flood queue size limit check (#857)

kcudnik committed:
[VS] Fix context config guid on RealObjectIdManager (#847)

Will help using multiple switches in VS with multiple contextes.

Pterosaur committed:
Add function to refresh macsec sa stat (#861)

Signed-off-by: Ze Gan <ganze718@gmail.com>

akokhan committed:
Fixed Popen args in physyncd_spawn() (#860)

Signed-off-by: Andriy Kokhan <andriyx.kokhan@intel.com>

dgsudharsan committed:
Tunnel counters #2

dgsudharsan committed:
Fixing build

kcudnik committed:
[meta] Allow objects with the same key exist on different switches (#865)

Signed-off-by: kcudnik <kcudnik@gmail.com>

kcudnik committed:
[syncd] Clean up log messages and add performance timer for bulk route create (#864)

gregshpit committed:
Ported Marvell armhf build on x86 for debian buster to use cross-compilation instead of qemu emulation (#852)

Current armhf Sonic build on amd64 host uses qemu emulation. Due to the nature of the emulation it takes a very long time, about 22-24 hours to complete the build. The change I did to improve the building time ports Sonic armhf build on amd64 host for Marvell platform for debian buster to use cross-compilation on arm64 host for armhf target. The overall Sonic armhf building time using cross-compilation is about 6 hours.

The Sonic configure and build for the armhf cross-compilation is as following:
NOJESSIE=1 NOSTRETCH=1 BLDENV=buster CROSS_BLDENV=1 make configure PLATFORM=marvell-armhf PLATFORM_ARCH=armhf
NOJESSIE=1 NOSTRETCH=1 BLDENV=buster CROSS_BLDENV=1 make target/sonic-marvell-armhf.bin

Sonic module should check if $CROSS_BUILD_ENVIRON is 'y' to make sure that it is cross-compilation build.

volodymyrsamotiy commented on syncd/FlexCounter.cpp:
```diff
@@ -1588,6 +1689,51 @@ void FlexCounter::collectBufferPoolCounters(
+
+        std::vector<uint64_t> tunnelStats(tunnelCounterIds.size());
+
+        // Get rif stats
```
I think here is a typo, should be ```tunnel``` but not ```rif```

dgsudharsan replied:
Modified

volodymyrsamotiy commented on syncd/FlexCounter.cpp:
```diff
@@ -2024,6 +2181,42 @@ void FlexCounter::updateSupportedBufferPoolCounters(
+    }
+
+    uint64_t value;
+    for (int cntr_id = SAI_TUNNEL_STAT_IN_OCTETS; cntr_id <= SAI_TUNNEL_STAT_OUT_PACKETS; ++cntr_id)
```
Is it possible to pass ```counterIds``` vector to this update function and iterate through it here instead of using hardcoded values?
Otherwise, if values here are hardcoded, it could cause some issues in case number of tunnel counters change.

For example it is done like that in ```updateSupportedQueueCounters()``` function.

dgsudharsan replied:
Updated

dgsudharsan committed:
Addressing code review comments

kcudnik committed:
[sairedis] Add context config fields conflict check (#874)

kcudnik committed:
Update .gitignore (#870)

kcudnik committed:
[meta] Use pragma once in serialize header (#873)

kcudnik committed:
[sairedis] Fix thread run flag on close (#872)

kcudnik committed:
[tests] Fix error code check (#871)

kcudnik committed:
[syncd] Fix status succes on bulk fallback (#867)

kcudnik committed:
[syncd] Fix access coldvid table per switch (#866)

Signed-off-by: kcudnik <kcudnik@gmail.com>

kcudnik committed:
[sairedis] Record create, remove and set response in sync mode (#869)

Start recording CRS operations response to sairedis.rec but only when response is not success.

wendani committed:
Clean codes (#876)

Signed-off-by: Wenda Ni <wonda.ni@gmail.com>

kcudnik committed:
[vslib] Clean Sai interface include (#880)

kcudnik committed:
Deprecate python module in favor to pyext (#882)

kcudnik committed:
[meta] Fix vlan set test (#879)

kcudnik committed:
[sairedis] Add comment api to recorder (#883)

Use new api to record hardware info when creating switch for easy read. Since hardware info attribute type is s8list, it will be serialized as array of integers, so to actually improve readability of actual value, put it as comment in recording file.

kcudnik committed:
[syncd] Add event name to timer watchdog (#868)

When watchdog thread will exceed 30sec it will print currently executing event like create/remove/get/set. This data can be also found in sairedis.rec, but this PR will also print this in syslog for convenience.

dgsudharsan committed:
Not modifying poll interval logic since it is handled in a different PR

dgsudharsan committed:
Merge branch 'master' of github.com:Azure/sonic-sairedis into evpn_vlan_tun_cntrs
